go-controller/pkg/kube/kube.go

```diff
@@ -387,3 +390,15 @@ func (k *KubeOVN) UpdateCloudPrivateIPConfig(cloudPrivateIPConfig *ocpcloudnetwo
 func (k *KubeOVN) DeleteCloudPrivateIPConfig(name string) error {
 	return k.CloudNetworkClient.CloudV1().CloudPrivateIPConfigs().Delete(context.TODO(), name, metav1.DeleteOptions{})
 }
+
+func (k *KubeOVN) UpdateEgressServiceStatus(namespace, name, host string) error {
```

Comment:
it is used in ClearServiceResources which doesn't have(or need) the CR. More specifically the node handler uses it and since we don't cache the CR I feel this is cleaner than caching it. wdyt?

Reply:
Would server side apply be something that we could leverage here? https://kubernetes.io/docs/reference/using-api/server-side-apply/

Additionally, is there a possibility for the CR to change while it is being processed in the sync function? This could lead to us updating the latest CR (not the one we processed). Not sure how probable this is, just food for thought.

Reply:
It could be better (not super-familiar with it), but even if stuff changes while we're syncing the resource it would be reconciled and reach the state the controller expects it to be in. I'd rather keep it the way it currently does and maybe tackle it later (in a non-refactoring pr) as it probably has implications on what I described in the other comment.

Reply:
yeah but iiuc to leverage resource versions we'll need to pass the object, but we don't have it in the node controller